src/net/tcp/incoming.rs

```diff
@@ -0,0 +1,30 @@
+use net::tcp::TcpListener;
+use net::tcp::TcpStream;
+
+use std::io;
+use futures::stream::Stream;
+use futures::{Poll, Async};
+
+/// Stream returned by the `TcpListener::incoming` function representing the
+/// stream of sockets received from a listener.
+#[must_use = "streams do nothing unless polled"]
+#[derive(Debug)]
+pub struct Incoming {
+    inner: TcpListener,
+}
+
+impl Incoming {
+    pub(crate) fn with_tcp_listener(listener: TcpListener) -> Incoming {
```

Comment:
Added to use in TcpListener:
```
    pub fn incoming(self) -> Incoming {
        Incoming::with_tcp_listener(self)
    }
```

Reply:
This function can probably just be called `new`.

Reply:
Done